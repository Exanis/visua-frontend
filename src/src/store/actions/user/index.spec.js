import * as Actions from './index';

describe('Users Actions', () => {
    it('getCurrentUserData', () => {
        expect(Actions.getCurrentUserData()).toMatchSnapshot();
    });
    it('logUserIn', () => {
        expect(Actions.logUserIn('username', 'password')).toMatchSnapshot();
    });
    it('onUserData', () => {
        expect(Actions.onUserData('test-user')).toMatchSnapshot();
    });
    it('onUserError', () => {
        expect(Actions.onUserError()).toMatchSnapshot();
    });
    it('onUserLoginSuccess', () => {
        expect(Actions.onUserLoginSuccess()).toMatchSnapshot();
    });
    it('onUserLoginError', () => {
        expect(Actions.onUserLoginError()).toMatchSnapshot();
    });
    it('onUserLoginClear', () => {
        expect(Actions.onUserLoginClear()).toMatchSnapshot();
    });
    it('logUserOut', () => {
        expect(Actions.logUserOut()).toMatchSnapshot();
    });
    it('updateUser', () => {
        expect(Actions.updateUser('test-uuid', 'test-firstname', 'test-lastname', 'test-email', 'test-password')).toMatchSnapshot();
    });
    it('onUpdateUserSuccess', () => {
        expect(Actions.onUpdateUserSuccess()).toMatchSnapshot();
    });
    it('onUpdateUserError', () => {
        expect(Actions.onUpdateUserError('error')).toMatchSnapshot();
    });
    it('resetCreateUserState', () => {
        expect(Actions.resetCreateUserState()).toMatchSnapshot();
    });
});