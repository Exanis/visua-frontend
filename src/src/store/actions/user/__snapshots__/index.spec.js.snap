// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Actions getCurrentUserData 1`] = `
Object {
  "type": "user.request",
}
`;

exports[`Users Actions logUserIn 1`] = `
Object {
  "type": "user.login",
  "user": Object {
    "password": "password",
    "username": "username",
  },
}
`;

exports[`Users Actions logUserOut 1`] = `
Object {
  "type": "user.logout",
}
`;

exports[`Users Actions onUpdateUserError 1`] = `
Object {
  "error": "error",
  "type": "user.update.error",
}
`;

exports[`Users Actions onUpdateUserSuccess 1`] = `
Object {
  "type": "user.update.success",
}
`;

exports[`Users Actions onUserData 1`] = `
Object {
  "type": "user.result",
  "user": "test-user",
}
`;

exports[`Users Actions onUserError 1`] = `
Object {
  "type": "user.error",
}
`;

exports[`Users Actions onUserLoginClear 1`] = `
Object {
  "type": "user.login.clear",
}
`;

exports[`Users Actions onUserLoginError 1`] = `
Object {
  "type": "user.login.error",
}
`;

exports[`Users Actions onUserLoginSuccess 1`] = `
Object {
  "type": "user.login.success",
}
`;

exports[`Users Actions resetCreateUserState 1`] = `
Object {
  "type": "user.update.clear",
}
`;

exports[`Users Actions updateUser 1`] = `
Object {
  "email": "test-email",
  "firstName": "test-firstname",
  "lastName": "test-lastname",
  "password": "test-password",
  "type": "user.update",
  "uuid": "test-uuid",
}
`;
